packages/strapi-design-system/src/Grid/Grid.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+
+import styled from 'styled-components';
+
+import { GridContext } from './GridContext';
+import { Box, BoxProps } from '../Box';
+import handleResponsiveValues, { ResponsiveValue } from '../helpers/handleResponsiveValues';
+
+export interface GridProps extends BoxProps {
+  gridCols?: number;
+  gap?: ResponsiveValue;
+}
+
+const GridWrapper = styled(Box)<Required<Pick<GridProps, 'gridCols' | 'gap'>>>`
+  display: grid;
+  grid-template-columns: repeat(${({ gridCols }) => gridCols}, 1fr);
+  ${({ theme, gap }) => handleResponsiveValues('gap', gap, theme)}
+`;
+
+export const Grid: React.FC<GridProps> = ({ gap = '0', gridCols = 12, ...props }) => {
+  const context = React.useMemo<Required<Pick<GridProps, 'gridCols' | 'gap'>>>(
```

Comment:
```suggestion
  const context = React.useMemo(
```

This should be inferred imo

Reply:
I thought the same, but it's not 😢 And it made me do that ugly thing